apps/condo/domains/billing/schema/ValidateQRCodeService.spec.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * @jest-environment node
+ */
+const index = require('@app/condo/index')
+const { faker } = require('@faker-js/faker')
+
+const { createInstance } = require('@open-condo/clients/address-service-client')
+const { setFakeClientMode, makeLoggedInAdminClient, expectToThrowGQLError } = require('@open-condo/keystone/test.utils')
+
+const {
+    createValidRuRoutingNumber,
+    createValidRuNumber,
+    createValidRuTin10,
+} = require('@condo/domains/banking/utils/testSchema/bankAccount')
+const { validateQRCodeByTestClient } = require('@condo/domains/billing/utils/testSchema')
+
+function generateQrCodeObj (extraAttrs = {}) {
+    const bic = createValidRuRoutingNumber()
+
+    return {
+        PersonalAcc: createValidRuNumber(bic),
+        PayeeINN: createValidRuTin10(),
+        PayerAddress: faker.address.streetAddress(true),
+        Sum: faker.random.numeric(6),
+        LastName: faker.random.alpha(10),
+        PaymPeriod: `${faker.datatype.number({ min: 1, max: 12 })}.${faker.datatype.number({
+            min: 2024,
+            max: 2099,
+        })}`,
+        BIC: bic,
+        PersAcc: faker.random.numeric(20),
+        ...extraAttrs,
+    }
+}
+
+function stringifyQrCode (qrCodeObj) {
+    return Buffer.from(Object.keys(qrCodeObj).reduce((qrStr, field) => {
+        return `${qrStr}|${field}=${qrCodeObj[field]}`
+    }, 'ST00012')).toString('base64')
+}
+
+describe('ValidateQRCodeService spec', () => {
+    setFakeClientMode(index)
```

Comment:
R we agreed with fakeclientmode usage? I've heard discussions that this is not the best option for writing tests

Reply:
For spec too?